PetGame/Services/ActivityService.cs

```diff
@@ -53,7 +53,7 @@ FROM Activity
                 cmd.Parameters.AddWithValue("@After", after);
                 cmd.Parameters.AddWithValue("@Limit", limit);
                 // if type is null, then insert a null. this will disable filtering by type
-                cmd.Parameters.AddWithValue("@Type", (object) (char)type ?? DBNull.Value);
+                cmd.Parameters.AddWithValue("@Type", (object) (char)type.GetValueOrDefault() ?? DBNull.Value);
```

Comment:
No, not `GetValueOrDefault()`, the `Value` property. https://docs.microsoft.com/en-us/dotnet/api/system.nullable-1.value?view=netframework-4.7.2#System_Nullable_1_Value